### PR TITLE
Log cmake and ninja outputs, and add more diagnostic messages to stdout.

### PR DESCRIPTION
**Pull request by sivachandra**

### CODE REVIEW

File: build_scripts/build_utils.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 OUT_DIR = os.path.join(PYTYPE_SRC_ROOT, "out")
 SRC_PYI_DIR = os.path.join(PYTYPE_SRC_ROOT, "pytype", "pyi")
 OUT_PYI_DIR = os.path.join(OUT_DIR, "pytype", "pyi")
+CMAKE_LOG = os.path.join(OUT_DIR, "cmake.log")
```
Comment (rchen152) [inline build_scripts/build_utils.py]: I've been wondering, is there a way to see the logs from a Travis run? I see /home/travis/... paths, but there's no obvious way to access them.

File: build_scripts/run_tests.py [hunk 2]
```diff
@@ -52,62 +56,81 @@ def print_report(self):
...
-    if mod_name:
-      print(l)
-    if log_file:
-      have_failures = True
-      if fail_collector:
-        fail_collector.add_failure(mod_name, log_file)
+  failed_targets = []
+  with open(build_utils.NINJA_LOG, "w") as ninja_log:
+    while True:
+      l = process.stdout.readline()
```
Comment (rchen152) [inline build_scripts/run_tests.py]: nit: lowercase L looks like number 1; maybe call this `line` or something?

Comment (sivachandra) [reply build_scripts/run_tests.py]: Done.

File: build_scripts/build_utils.py (hunk 1 above)
Comment (sivachandra) [reply build_scripts/build_utils.py]: As discussed offline, we should setup uploading build artifacts to GCS so that we can download and see the logs.

**Pull request merged**